backend/src/main/java/gov/cdc/usds/simplereport/service/DiseaseService.java

```diff
@@ -26,50 +29,42 @@ public class DiseaseService {
 
   private final SupportedDiseaseRepository _supportedDiseaseRepo;
 
-  private SupportedDisease covid;
-  private SupportedDisease fluA;
-  private SupportedDisease fluB;
-
-  private final Map<UUID, SupportedDisease> supportedDiseaseMap = new HashMap<>();
-
   public void initDiseases() {
-    covid = _supportedDiseaseRepo.findByName(COVID19_NAME).orElse(null);
-    fluA = _supportedDiseaseRepo.findByName(FLU_A_NAME).orElse(null);
-    fluB = _supportedDiseaseRepo.findByName(FLU_B_NAME).orElse(null);
-
-    Optional.ofNullable(covid).ifPresent(sd -> supportedDiseaseMap.put(sd.getInternalId(), sd));
-    Optional.ofNullable(fluA).ifPresent(sd -> supportedDiseaseMap.put(sd.getInternalId(), sd));
-    Optional.ofNullable(fluB).ifPresent(sd -> supportedDiseaseMap.put(sd.getInternalId(), sd));
+    getKnownSupportedDiseasesMap();
   }
 
   public List<SupportedDisease> fetchSupportedDiseases() {
-    return _supportedDiseaseRepo.findAll();
+    return getDiseaseNameToSupportedDiseaseMap().values().stream().toList();
   }
 
+  @Cacheable(SUPPORTED_DISEASE_ID_MAP)
   public Map<UUID, SupportedDisease> getKnownSupportedDiseasesMap() {
-    return supportedDiseaseMap;
+    return _supportedDiseaseRepo.findAll().stream()
+        .collect(Collectors.toMap(SupportedDisease::getInternalId, Function.identity()));
+  }
+
+  public Map<String, SupportedDisease> getDiseaseNameToSupportedDiseaseMap() {
+    return getKnownSupportedDiseasesMap().values().stream()
+        .collect(Collectors.toMap(SupportedDisease::getName, Function.identity()));
   }
 
   public SupportedDisease getDiseaseByName(String name) {
-    return switch (name) {
-      case COVID19_NAME -> covid;
-      case FLU_A_NAME -> fluA;
-      case FLU_B_NAME -> fluB;
-      default -> _supportedDiseaseRepo
-          .findByName(name)
-          .orElseThrow(() -> new IllegalArgumentException("Disease not found"));
-    };
+    return getDiseaseNameToSupportedDiseaseMap().get(name);
   }
 
   public SupportedDisease covid() {
-    return covid;
+    return getDiseaseByName(COVID19_NAME);
   }
 
   public SupportedDisease fluA() {
-    return fluA;
+    return getDiseaseByName(FLU_A_NAME);
   }
 
   public SupportedDisease fluB() {
-    return fluB;
+    return getDiseaseByName(FLU_B_NAME);
+  }
+
+  public SupportedDisease rsv() {
```

Comment:
I opted to leave these in rather than replace them with the corresponding `getDiseaseByName(DISEASE_NAME)` method in all the callers since I think having the disease-specific methods like we do here is clearner, but if we think that's preferable, I can remove those as well.